shops/forms.py

```diff
@@ -25,3 +28,38 @@ def is_valid(self):
         return True
 
 
+class WeeklySummaryForm(forms.ModelForm):
+    class Meta:
+        model = models.WeeklyShopSummary
+        fields = ['upload']
+
+    # validation function
+    def clean_upload(self):
+        # Checking if file type and format is corrext
+        name = self.cleaned_data['upload'].name
+        if '.xlsx' not in name:
+            raise ValidationError('Invalid file type - data must be of type .xlsx')
+        name = name.replace('.xlsx', '')
+        r = re.compile('.{4}-.{2}-.{2}-.{4}')
+        if len(name) != 16 or (not r.match(name)):
+            raise ValidationError('File name must be of format \'yyyy-mm-dd-shopcode\'')
+        date, shop_code = name.rsplit('-', 1)
+        try:
+            date_object = timezone.make_aware(datetime.strptime(date, '%Y-%m-%d'))
```

Comment:
Everything in Python is an object (even the datetime.date function ;-) )

You could use one of the following:

- date
- date_
- upload_date
- week_date
- monday_date

Ah, I just realised that you've already used date. In that case you could use 'raw_date' or 'date_as_string' for the first date and then 'date' for the 'proper' date. I regularly use the 'raw_' prefix for something which isn't in the right shape yet - if only to warn myself that I shouldn't use this anywhere else